hazelcast/src/main/java/com/hazelcast/map/IMap.java

```diff
@@ -2978,4 +2980,29 @@ default void addIndex(IndexType type, String... attributes) {
      */
     V computeIfAbsent(@Nonnull K key, @Nonnull Function<? super K, ? extends V> mappingFunction);
 
+    /**
+     * {@inheritDoc}
+     *
+     * <p> </p>
+     * <p>
+     *     If the supplied {@code action} is a lambda, anonymous class or an inner class,
+     *     it would be executed locally. Same would happen if it is not serializable.
+     *     This may result in multiple round-trips between hazelcast nodes.
+     *</p>
+     * <p>
+     *     Otherwise (i.e. if it is a top-level class or a member class, and it is serializable), the function <i>may be</i> sent
+     *     to the servers which own the partitions/keys. This results in a much less number of remote calls.
+     *     Note that in this case, side effects of the {@code action} may not be visible to the local JVM
+     *     If users intend to install the changed value in the map entry, the {@code action} must do so itself (e.g. by
```

Comment:
In this case, I'd propose the user to simply use `IMap#executeOnEntries(com.hazelcast.map.EntryProcessor<K,V,R>)` directly, since it can change the value by calling `java.util.Map.Entry#setValue` from within the entry processor.